pkg/build/nodeimage/internal/kube/builder_remote.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/kind/pkg/log"
+)
+
+type remoteBuilder struct {
+	version string
+	logger  log.Logger
+}
+
+var _ Builder = &remoteBuilder{}
+
+func NewRemoteBuilder(logger log.Logger, version string) (Builder, error) {
+	return &remoteBuilder{
+		version: version,
+		logger:  logger,
+	}, nil
+}
+
+// Build implements Bits.Build
+func (b *remoteBuilder) Build() (Bits, error) {
+	url := "https://dl.k8s.io/" + b.version + "/kubernetes-server-linux-amd64.tar.gz"
```

Comment:
if we make this parmetizable as default `https://dl.k8s.io/"` but that people can define their own url or `file:://home/aojea/mybuild` we don't need the directory, people with custom builds just have to create the tarball as we expect (with current approach it has to do it anyway, as it has to have the same directory layout

Reply:
i am holding off on the parameterizable url and dedicating this to JUST our released artifacts as I did add another option to use tgz from file on disk.